keras_core/backend/__init__.py

```diff
@@ -6,6 +6,21 @@
     # upon import.
     import torch
 
+if backend() != "tensorflow":
+    import tensorflow as tf
```

Comment:
As a general policy we should only import TF when requested. Right now it gets imported lazily the first time it's needed, in `utils/module_utils.py`. We should customize the `initialize()` method for TF to insert this routine.

Reply:
Hmm, I think in this case, we may be better off switching this to just setting `os.environ["TF_FORCE_GPU_ALLOW_GROWTH"] = "true"` in `config.py`. This is essentially how colab handles the problem.

The issue with tf become a delayed import is that a user script might import tensorflow first, not use the lazy model, and lead to OOMs outside of our control. Setting the environment variable is lightweight, we can do it first thing, and it will not affect tf until it's imported. What do you think?

Reply:
> Hmm, I think in this case, we may be better off switching this to just setting

SGTM. Lightweight indeed and the Colab precedent shows it's fine.

To note, I have made the changes described above -- now we only import TF if using KPL or if saving to GCS. But many users will likely import it anyway (e.g. tf.data  + JAX)